clientEncryptionToCPKNMigrationSamples/ClientSideKeyVaultKeyToCustomerManagedKey/src/main/java/Decrypt.java

```diff
@@ -0,0 +1,130 @@
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.cryptography.AsyncKeyEncryptionKey;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.keys.KeyClient;
+import com.azure.security.keyvault.keys.KeyClientBuilder;
+import com.azure.security.keyvault.keys.cryptography.KeyEncryptionKeyClientBuilder;
+import com.azure.security.keyvault.keys.cryptography.models.KeyWrapAlgorithm;
+import com.azure.security.keyvault.keys.models.KeyVaultKey;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobClientBuilder;
+import com.azure.storage.blob.specialized.cryptography.EncryptedBlobClient;
+import com.azure.storage.blob.specialized.cryptography.EncryptedBlobClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Downloads and decrypts blob using key vault, then reuploads using customer-managed keys through encryption scopes
+ */
+public class Decrypt {
```

Comment:
I think Decrypt may be better named Main or something to identify that this is the core logic of the sample

Reply:
Migration? That seems to be what Carmen used